### DISCUSSION TITLE
Where can I find the test suite input and reference output files?

### DISCUSSION THREAD

**Discussion by jajberni** [KUL-RSDA/AquaCrop#322] [Q&A] [answered]
I have successfully compiled AquaCrop in MacOS, but when trying to run the tests, there are no input and reference output files to test. It would be great to include them in the repo or at least a download link for downloading a folder to point to with $AQUACROP_TEST_ROOT. 

Other than that, congratulations on the outstanding achievement of upgrading AquaCrop and porting it to Fortran!

**Comment by mbechtold** [comment, marked as answer]
Good to hear. Thx for the feedback.
The v7.0 release includes a testcase, you find it here:
https://github.com/KUL-RSDA/AquaCrop/releases/tag/v7.0

**Reply by jajberni** [reply]
Thank you @mbechtold!

However, when I extract the zip file and try to run the tests, test_europe runs fine, but `test_harvest` and `test_perennial`fail.

According to the code, it looks for the folder with those names in the `AQUACROP_TEST_ROOT` folder:

```
test_dir = os.path.join(os.environ['AQUACROP_TEST_ROOT'], 'test_perennial')
```

However, these folders don't exist in this zip file. 

I think that am missing something here.

**Reply by mbechtold** [reply]
Oh, you are trying to run the full test suite which we used during code conversion to test for zero-diff. The input for this is not provided on github. That's too much data. We use this only internally. 
I was referring to the testcase zip folder in the release:
https://github.com/KUL-RSDA/AquaCrop/releases/download/v7.0/aquacrop-7.0-testcase.zip
There is a README in this zip file with all information.

**Reply by jajberni** [reply]
Interestingly, running the tests passes for test_europe, even if each log contains the following:

```
========= 0 0 False =========
SKIP: missing soil file for 0_0: testcase/test_europe/INPUT/soil_files/0_0.SOL
```

The condition here:
https://github.com/KUL-RSDA/AquaCrop/blob/b36e30745c08a848113b84e7a6454eda8784b7e9/tests/test_europe.py#L106 

Returns without giving a test error, leading to confusion when running automatic tests.

**Reply by jajberni** [reply]
> Oh, you are trying to run the full test suite which we used during code conversion to test for zero-diff. The input for this is not provided on github. That's too much data. We use this only internally. I was referring to the testcase zip folder in the release: https://github.com/KUL-RSDA/AquaCrop/releases/download/v7.0/aquacrop-7.0-testcase.zip There is a README in this zip file with all information.

Thank you, it makes sense now! I was assuming that the automatic tests would run for any developer. I tried the testcase and it runs great.
